proxy-socket-core/src/main/java/net/airvantage/proxysocket/core/v2/ProxyProtocolV2Decoder.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * MIT License
+ * Copyright (c) 2025 Semtech
+ */
+package net.airvantage.proxysocket.core.v2;
+
+import net.airvantage.proxysocket.core.ProxyProtocolParseException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Dependency-free Proxy Protocol v2 utilities (validate/parse/build).
+ */
+public final class ProxyProtocolV2Decoder {
+    private ProxyProtocolV2Decoder() {}
+
+    private static final byte[] SIG = "\r\n\r\n\0\r\nQUIT\n".getBytes(StandardCharsets.ISO_8859_1);
+    private static final int IPV4_ADDR_LEN = 4;
+    private static final int IPV6_ADDR_LEN = 16;
+    private static final int UNIX_ADDR_LEN = 216;
+    private static final int PORT_LEN = 2;
+    private static final int TLV_HEADER_LEN = 3;
+
+    public static ProxyHeader parse(byte[] data, int offset, int length) throws ProxyProtocolParseException {
+        if (data == null || offset < 0 || length < 0) throw new ProxyProtocolParseException("Null data");
+        if ((length+offset) > data.length) throw new ProxyProtocolParseException("Invalid offset/length");
+
+        int end = offset + length;
+        if (SIG.length + 4 > length) throw new ProxyProtocolParseException("Insufficient data for header");
+
+        for (int i = 0; i < SIG.length; i++) {
+            if (data[offset + i] != SIG[i]) {
+                throw new ProxyProtocolParseException("Invalid signature");
+            }
+        }
+
+        int pos = offset + SIG.length;
+
+        int verCmd = data[pos++] & 0xFF; // version/command
+        int version = (verCmd >> 4) & 0x0F;
+        if (version != 2) throw new ProxyProtocolParseException("Invalid version");
+        int cmd = verCmd & 0x0F;
+        ProxyHeader.Command command = cmd == 0x00 ? ProxyHeader.Command.LOCAL : ProxyHeader.Command.PROXY;
```

Comment:
Why not just something like 
```suggestion
        int verCmd = data[pos++] & 0xFF; // version/command
        int version = (verCmd >> 4) & 0x0F;
        if (version != 2) throw new ProxyProtocolParseException("Invalid version");
        int cmd = verCmd & 0x0F;
        // parse command
        ProxyHeader.Command command;
        switch(cmd){
        case  0x00 : 
           command = ProxyHeader.Command.LOCAL;
           break;
         case  0x01 : 
            command = ProxyHeader.Command.PROXY;
            break;
           default : 
              throw new ProxyProtocolParseException("Invalid command");
        }
```
Probably even better to but that switch in a dedicated method.

I didn't test that code